crates/evm/src/fuzz/invariant/executor.rs

```diff
@@ -428,6 +428,38 @@ impl<'a> InvariantExecutor<'a> {
             ["targetSenders", "excludeSenders", "targetContracts", "excludeContracts"]
                 .map(|method| self.get_list::<Address>(invariant_address, abi, method));
 
+        let interfaces =
+            self.get_list::<(Address, Vec<String>)>(invariant_address, abi, "targetInterfaces");
+
+        // `combined` is a BTreeMap that will store contracts mapped by their target address.
+        // This map is used to merge functions of the specified interfaces for the same address.
+        let mut combined: TargetedContracts = BTreeMap::new();
+
+        // Loop through each address and its associated artifact identifiers.
+        // We're borrowing the interfaces here to avoid taking full ownership of them.
+        for (key, identifiers) in &interfaces {
```

Comment:
All of this is done for being able to use interfaces/proxies right? Could we add a little bit of docs on how this helps achieve this usage? without context it's hard to understand that this is for diamong proxies

Reply:
Effectively making the address known without having to deploy the contract during setUp. Will expand on the docs to include this info.

The issue is not really just diamond proxies but really just interface based fuzzing. I develop this I realized the use case is a lot more aligned for fork based fuzzing, or even fuzzing Vyper contracts. Diamond and Router proxies is just the use case I started with.